aioxmpp/avatar/service.py

```diff
@@ -549,24 +729,28 @@ def publish_avatar_set(self, avatar_set):
         avatar metadata set in pubsub. The `avatar_set` must be an
         instance of :class:`AvatarSet`.
         """
-        yield from self._check_for_pep()
-
         id_ = avatar_set.png_id
 
         with (yield from self._publish_lock):
-            yield from self._pubsub.publish(
-                None,
-                namespaces.xep0084_data,
-                avatar_xso.Data(avatar_set.image_bytes),
-                id_=id_
-            )
+            try:
+                yield from self._pep.publish(
+                    namespaces.xep0084_data,
+                    avatar_xso.Data(avatar_set.image_bytes),
+                    id_=id_
+                )
 
-            yield from self._pubsub.publish(
-                None,
-                namespaces.xep0084_metadata,
-                avatar_set.metadata,
-                id_=id_
-            )
+                yield from self._pep.publish(
+                    namespaces.xep0084_metadata,
+                    avatar_set.metadata,
+                    id_=id_
+                )
+            finally:
+                if self._synchronize_vcard:
```

Comment:
I don’t think we should try to push to the vcard if the push to the PEP failed.

Reply:
What if our server only support vcard?

Reply:
Right, good point. In that case, we need to try both, so using asyncio.gather/asyncio.wait seems reasonable.